tests/posix/common/src/pthread_attr.c

```diff
@@ -0,0 +1,638 @@
+
+	/* degenerate cases */
+	{
+		if (false) {
+			/* undefined behaviour */
+			zassert_equal(pthread_attr_setschedpolicy(NULL, SCHED_INVALID), EINVAL);
+			zassert_equal(pthread_attr_setschedpolicy(NULL, policy), EINVAL);
+			zassert_equal(
+				pthread_attr_setschedpolicy((pthread_attr_t *)&uninit_attr, policy),
+				EINVAL);
+		}
+		zassert_equal(pthread_attr_setschedpolicy(&attr, SCHED_INVALID), EINVAL);
+	}
+
+	zassert_ok(pthread_attr_setschedpolicy(&attr, SCHED_OTHER));
+	/* read back the same policy we just wrote */
+	policy = SCHED_INVALID;
+	zassert_ok(pthread_attr_getschedpolicy(&attr, &policy));
+	zassert_equal(policy, SCHED_OTHER);
+
+	can_create_thread(&attr);
+}
+
+ZTEST(pthread_attr, test_pthread_attr_getstack)
+{
+	void *stackaddr = (void *)BIOS_FOOD;
+	size_t stacksize = BIOS_FOOD;
+
+	/* degenerate cases */
+	{
+		if (false) {
+			/* undefined behaviour */
+			zassert_equal(pthread_attr_getstack(NULL, NULL, NULL), EINVAL);
+			zassert_equal(pthread_attr_getstack(NULL, NULL, &stacksize), EINVAL);
+			zassert_equal(pthread_attr_getstack(NULL, &stackaddr, NULL), EINVAL);
+			zassert_equal(pthread_attr_getstack(NULL, &stackaddr, &stacksize), EINVAL);
+			zassert_equal(pthread_attr_getstack(&uninit_attr, &stackaddr, &stacksize),
+				      EINVAL);
+		}
+		zassert_equal(pthread_attr_getstack(&attr, NULL, NULL), EINVAL);
+		zassert_equal(pthread_attr_getstack(&attr, NULL, &stacksize), EINVAL);
+		zassert_equal(pthread_attr_getstack(&attr, &stackaddr, NULL), EINVAL);
+	}
+
+	zassert_ok(pthread_attr_getstack(&attr, &stackaddr, &stacksize));
+	zassert_not_equal(stackaddr, (void *)BIOS_FOOD);
+	zassert_not_equal(stacksize, BIOS_FOOD);
+}
+
+ZTEST(pthread_attr, test_pthread_attr_setstack)
+{
+	void *stackaddr;
+	size_t stacksize;
+	void *new_stackaddr;
+	size_t new_stacksize;
+
+	/* valid values */
+	zassert_ok(pthread_attr_getstack(&attr, &stackaddr, &stacksize));
+
+	/* degenerate cases */
+	{
+		if (false) {
+			/* undefined behaviour */
+			zassert_equal(pthread_attr_setstack(NULL, NULL, 0), EACCES);
+			zassert_equal(pthread_attr_setstack(NULL, NULL, stacksize), EINVAL);
+			zassert_equal(pthread_attr_setstack(NULL, stackaddr, 0), EINVAL);
+			zassert_equal(pthread_attr_setstack(NULL, stackaddr, stacksize), EINVAL);
+			zassert_equal(pthread_attr_setstack((pthread_attr_t *)&uninit_attr,
+							    stackaddr, stacksize),
+				      EINVAL);
+		}
+		zassert_equal(pthread_attr_setstack(&attr, NULL, 0), EACCES);
+		zassert_equal(pthread_attr_setstack(&attr, NULL, stacksize), EACCES);
+		zassert_equal(pthread_attr_setstack(&attr, stackaddr, 0), EINVAL);
+	}
+
+	/* ensure we can create and join a thread with the default attrs */
+	can_create_thread(&attr);
+
+	/* set stack / addr to the current values of stack / addr */
+	zassert_ok(pthread_attr_setstack(&attr, stackaddr, stacksize));
+	can_create_thread(&attr);
+
+	/* qemu_x86 seems to be unable to set thread stacks to be anything less than 4096 */
+	if (!IS_ENABLED(CONFIG_X86)) {
```

Comment:
@ceolin or perhaps @dcpleung - hoping you can do a review here.

This seems kind of weird to me. The thread stack that is allocated (IIRC via pool) is 4096 bytes (via `K_THREAD_STACK_ARRAY_DEFINE()`). If I subtract 1 from that size, this section breaks.

I know that for statically allocated thread stacks, a smaller size is fine for x86. Does this have to do with MMU / page alignment?